dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -527,9 +527,11 @@ HRESULT WINAPI CRecycleBin::CompareIDs(LPARAM lParam, PCUIDLIST_RELATIVE pidl1,
             diff = pData1->FileSize.QuadPart - pData2->FileSize.QuadPart;
             return MAKE_COMPARE_HRESULT(diff);
         case 4: /* Type */
-            pName1 = PathFindExtensionW(pData1->szName);
-            pName2 = PathFindExtensionW(pData2->szName);
-            result = wcsicmp(pName1, pName2);
+            SHELLDETAILS Details1;
+            SHELLDETAILS Details2;
+            GetDetailsOf(pidl1, COLUMN_TYPE, &Details1);
+            GetDetailsOf(pidl2, COLUMN_TYPE, &Details2);
+            result = stricmp(Details1.str.cStr, Details2.str.cStr);
```

Comment:
Not a good way of doing that! What about the ANSI vs. UNICODE stuff here?
See instead https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/shell32/folders/CFSFolder.cpp;hb=6623b8d15527551e044da01d719209ed6effa552#l975

Reply:
Yeah, but there it compares extensions, i'm comparing the typenames, and GetDetailsOf returns the typename on ANSI style, so there is no need of conversion

Reply:
The typenames are localized, that's why this won't generally do a good conversion. Also, when you sort per type on windows, it sorts it by extension really, not by the alphabetical displayed type name.

Reply:
then, i will revert this specific change

Reply:
Done